pkg/controller/migplan/validation.go

```diff
@@ -654,3 +665,79 @@ func containsAccessMode(modeList []kapi.PersistentVolumeAccessMode, accessMode k
 	}
 	return false
 }
+
+type NamespacePods struct {
+	Namespace string         `json:"namespace"`
+	Pods      []PodWithPhase `json:"pods"`
+}
+
+type PodWithPhase struct {
+	Name   string        `json:"name"`
+	Status kapi.PodPhase `json:"status"`
+}
+
+// Validate the pods, all should be healthy befor migration
+func (r ReconcileMigPlan) validatePods(plan *migapi.MigPlan) error {
+	cluster, err := plan.GetSourceCluster(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	if cluster == nil || !cluster.Status.IsReady() {
+		return nil
+	}
+
+	client, err := cluster.GetClient(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	failedPodSlice := make([]NamespacePods, 0)
+
+	for _, ns := range plan.Spec.Namespaces {
+		options := k8sclient.InNamespace(ns)
+		podList := kapi.PodList{}
+		err := client.List(context.TODO(), options, &podList)
+		if err != nil {
+			log.Trace(err)
+			return err
+		}
+
+		if len(podList.Items) == 0 {
+			continue
+		}
+
+		podPhaseSlice := make([]PodWithPhase, 0)
+
+		for _, pod := range podList.Items {
+			if pod.Status.Phase == kapi.PodRunning || pod.Status.Phase == kapi.PodSucceeded {
+				continue
+			}
+			podPhaseSlice = append(podPhaseSlice, PodWithPhase{Name: pod.Name, Status: pod.Status.Phase})
```

Comment:
The `Condition` supports and `Items` field that is designed to make this easy.
Suggest:
```
for _, pod := range podList.Items {
    if pod.Status.Phase == kapi.PodRunning || pod.Status.Phase == kapi.PodSucceeded {
        continue
    }
    condition := plan.Status.FindCondition(SourcePodsNotHealthy)
    if condition == nil {
        condition = migapi.Condition{
            Type:     SourcePodsNotHealthy,
            Status:   True,
            Reason:   NotHealthy,
            Category: Warn,
            Message:  "Pods not healthy on the source cluster [].",
            Items: []string{}
        }
        plan.Status.SetCondition(condition)
    }
    ref := fmt.Sprintf("%s:%s", pod.Namespace, pod.Name)
    condition.Items = append(condition.Items, ref)
}
```

Reply:
@jortel Your suggestion was my initial solution, but I removed it later. I want the message to be readable for UI. An array of JSON objects - `{"namespace": ..., "pods": []}`  is much more easy to parse in JS than grouping namespaces and pods from an array of strings [namespace1:pod1, namespace1:pod2, namespace2:pod2].

Reply:
While appreciate your intention - according to kubernetes API conventions, The `Message` should be a _"human-readable message indicating details about last transition"_ not machine parsable data.  I don't think a big hunk of json is particularly readable.  Also, the API is intended for general use and not just to support the UI.  This approach deviates from all other `MIG` conditions.

Has the UI expressed a requirement to parse the list of pods?

[1] https://github.com/eBay/Kubernetes/blob/master/docs/devel/api-conventions.md#typical-status-properties

Reply:
@jwmatthews @eriknelson Hi, can you take a look at this ^^ ? What's more desired approach from UI point of view?

Reply:
I commented in the google doc about this as well, I'm inclined to agree with Jeff here. The message field inside of a condition should be human readable, I liked the approach you mentioned in the google doc around having a separate field on the status (`verify` or something) where this JSON can be stored and read from. Curious what others think about this

Reply:
@jortel @dymurray I've just replaced the JSON with suggested error message:)